components/UserFormReactHookForm.tsx

```diff
@@ -37,13 +43,23 @@ const UserFormReactHookForm = () => {
             className={styles.input}
             {...register("firstName", { required: true, minLength: 2 })}
           />
+          {errors.firstName && (
+            <p className={styles.error}>
+              First name longer than two characters is required
+            </p>
+          )}
         </div>
         <div className={styles.labelInputContainer}>
```

Comment:
Inputs in UserFormReactHookForm should also be extracted into separate components. See https://react-hook-form.com/api/useformcontext

Reply:
done